controllers/controller/vpnkey_rotation_controller_test.go

```diff
@@ -0,0 +1,298 @@
+			}, timeout, interval).Should(BeTrue())
+
+			// check if expiry TS is not zero
+			Eventually(func() bool {
+				err := k8sClient.Get(ctx, getKey, createdVpnKeyConfig)
+				Expect(err).To(BeNil())
+				return !createdVpnKeyConfig.Spec.CertificateExpiryTime.IsZero()
+			}, timeout, interval).Should(BeTrue())
+		})
+
+		It("Should recreate/retrigger jobs for cert creation once it expires", func() {
+			createdVpnKeyConfig := &v1alpha1.VpnKeyRotation{}
+
+			err := k8sClient.Get(ctx, types.NamespacedName{
+				Namespace: slice.Namespace,
+				Name:      slice.Name,
+			}, createdVpnKeyConfig)
+
+			Expect(err).To(BeNil())
+
+			// expire it
+			time := time.Now().Add(-1 * time.Hour)
+			createdVpnKeyConfig.Spec.CertificateExpiryTime = &metav1.Time{Time: time}
+			err = k8sClient.Update(ctx, createdVpnKeyConfig)
+			Expect(err).To(BeNil())
+
+			// expect new jobs to be created
+			job := batchv1.JobList{}
+			o := map[string]string{
+				"SLICE_NAME": sliceName,
+			}
+			listOpts := []client.ListOption{
+				client.MatchingLabels(
+					o,
+				),
+			}
+			Eventually(func() bool {
+				err = k8sClient.List(ctx, &job, listOpts...)
+				if err != nil {
+					return false
+				}
+				return len(job.Items) > 0
+			}, timeout, interval).Should(BeTrue())
+		})
+		// cluster onboarding tests
+		It("Should Update VPNKeyRotation Config in case a new cluster is added", func() {
+			// update sliceconfig
+			createdSliceConfig := &v1alpha1.SliceConfig{}
+
+			err := k8sClient.Get(ctx, types.NamespacedName{
+				Namespace: slice.Namespace,
+				Name:      slice.Name,
+			}, createdSliceConfig)
+
+			Expect(err).To(BeNil())
+			createdSliceConfig.Spec.Clusters = append(createdSliceConfig.Spec.Clusters, "worker-3")
+			Expect(k8sClient.Update(ctx, createdSliceConfig)).Should(Succeed())
+
+			createdVpnKeyConfig := &v1alpha1.VpnKeyRotation{}
+
+			Eventually(func() []string {
+				err = k8sClient.Get(ctx, types.NamespacedName{
+					Namespace: slice.Namespace,
+					Name:      slice.Name,
+				}, createdVpnKeyConfig)
+
+				if err != nil {
+					return []string{""}
+				}
+				return createdVpnKeyConfig.Spec.Clusters
+			}, timeout, interval).Should(Equal([]string{"worker-1", "worker-2", "worker-3"}))
+		})
+		It("Should Update Cluster(3) Gateway Mapping", func() {
+			createdVpnKeyConfig := &v1alpha1.VpnKeyRotation{}
+			getKey := types.NamespacedName{
+				Namespace: slice.Namespace,
+				Name:      slice.Name,
+			}
+			Eventually(func() bool {
+				err := k8sClient.Get(ctx, getKey, createdVpnKeyConfig)
+				Expect(err).To(BeNil())
+				return createdVpnKeyConfig.Spec.ClusterGatewayMapping != nil
+			}, timeout, interval).Should(BeTrue())
+
+			// the length of map should be 3
+			Eventually(func() bool {
+				err := k8sClient.Get(ctx, getKey, createdVpnKeyConfig)
+				Expect(err).To(BeNil())
+				return len(createdVpnKeyConfig.Spec.ClusterGatewayMapping) == 3
+			}, timeout, interval).Should(BeTrue())
```

Comment:
Add the validation for ClusterGatewayMapping map

Reply:
Not sure what does that mean?